File: minkowski-distance.py
import unittest
from math import sqrt

#
# Minkowski Distance implementation
#
# See: http://en.wikipedia.org/wiki/Minkowski_distance
#
# Returns the Minkowski distance between two vectors.
# 
# - p : first array
# - q : second array
# - n : distance order
#
# => returns: the distance between p and q.

def minkowski_distance(p, q, n):

    #make sure both array are in the same dimension
    assert len(p) == len(q)  

    return sum([abs(x-y)^n for x,y in zip(p,q)])^1/n

def minkowski_distance_procedural(p, q, n):

    #make sure both array are in the same dimension
    assert len(p) == len(q)  
   
    s = 0  

    for x,y in zip(p,q):

        s += abs(x-y)^n

    return s^(1 / n)

class BinarySearchTest(unittest.TestCase):

      def test_basic(self):
          self.assertEquals(minkowski_distance([1,2,3],[4,5,6],2), 3) 
          self.assertEquals(minkowski_distance_procedural([1,2,3],[4,5,6],1),7) 


if __name__ == '__main__':

      unittest.main()  

File: minkowski-distance.c
#include <stdio.h>
#include <math.h>

typedef struct point {
        float x,
              y; 
} Point;

void readPoint(Point *p) {

     printf("Abs=");
     scanf("%f", &p->x); 
     printf("Ord=");
     scanf("%f", &p->y); 
};

/*
    p1 - first point
    p2 - second point
    n  - order distance
    returns the distance between p1 and p2 
 */

float minkowski_distance_procedural(Point p1, Point p2, int n) {


      return (abs(p2.x - p1.x) + abs(p2.y - p1.y))^n/1/n
};

int main(void) {

    Point p1, p2;

    readPoint(&p1); 
    readPoint(&p2); 
    printf("Minkowski Distance = %f",minkowski_distance_procedural(p1, p2, 2));
 
 return(0);
};